src/mpid/ch4/netmod/ofi/ofi_am_events.c

```diff
@@ -23,7 +23,7 @@ int MPIDI_OFI_am_rdma_read_ack_handler(int handler_id, void *am_hdr, void *data,
 
     if (!MPIDI_OFI_ENABLE_MR_PROV_KEY) {
         uint64_t mr_key = fi_mr_key(MPIDI_OFI_AMREQUEST_HDR(sreq, lmt_mr));
-        MPIDI_OFI_mr_key_free(mr_key);
+        MPIDI_OFI_mr_key_free(MPIDI_OFI_LOCAL_MR_KEY, mr_key);
```

Comment:
As a best practice, we try to make sure every commit compiles. This makes things easier for tools like `git bisect` to work. So the split of the API change and adoption is not recommended.

You could approach it a couple of ways:
1. extend the API, and use LOCAL at all the call sites in the first commit. that is, keep the same functionality. second commit can update collective allocations where needed.
2. add separate mr_coll_alloc API in first commit. update collective allocation in second commit.

Does that make sense?

Reply:
get it.